### ISSUE TITLE
[Bug] Editor. sets the system default editor, when leaving its settings

### ISSUE THREAD

**Issue by Olf0** [GoAlexander/harbour-editor#94]
Issue: Editor. sets the system default editor (to either Editor. or Jolla Notes), when leaving the settings. 

Expected behaviour: Editor. does not change the MIME types handling, when nothing is changed in its settings. 

Workaround: Use [Mimer](https://openrepos.net/content/llelectronics/mimer) to set the MIME type handlers to any installed application, each time after leaving Editor.'s settings.

Suggested solution: Eliminate the setting  "Default text editor".  It is superfluous, as this is what MIMER (from Openrepos.net) is for (but in a far more general manner). 

Note:
IMO leaving the settings page should only write altered settings to persistent storage, in general (i.e. regardless which setting(s)).  
Or (a little less elegant) write a setting to storage, right after it has been changed; then there is no need to write anything when the settings page is closed.

Tested with Editor. v0.8 on a Jolla 1 phone under SailfishOS 2.1.0.11

**Comment by GoAlexander** [comment]
Did you swipe back to editor space ? 
Or close the app?

**Comment by GoAlexander** [comment]
You have to swipe back to the editor page to save the settings

**Comment by Olf0** [comment]
I sure swiped back (or, to be precise: used the button in the upper left corner), because otherwise changed settings are not set. 

But that was not the point!

Imagine following setup (which was mmin until last week), for example: 
- You love editors, so you try every native editor on your SailfishOS phone (plus some Android editors under AlienDalvik).  But you are also aware of the risks, which young and rapidly developing software may bring with every new release: Bugs in new features, regressions in old ones, sudden UI changes ( ;) ) etc. 
- Hence you have set the default editor (with MIMER) to the old, trusted TinyEdit, which was the only usable native editor anyway, before Editor. and tIDEditor were created. 
- Now, each time you change something in Editor.'s settings, as a side effect TinyEdit is not default editor, anymore.  So changing a settings of Editor. means for you to set the default editor anew with MIMER.  (You can replace TinyEdit by any other editor, including Android ones, and still this example holds true.)
- If one wants Editor. as default editor, MIMER should be used! 

In short: Please eliminate the "Default editor" setting in Editor.'s settings completely, as it is superfluous *and* doing the wrong thing, as it does not provide a choice for setting *any* installed editor as default editor (like MIMER does), but only for Editor. and (the unusable) Notes. 

You may implement a button "Set Editor. as default editor", which only does something, when explicitly clicked. 
Or just leave this task (of setting default apps) to MIMER, which is made just for that.  IMHO, that the easiest and best resolution.

**Comment by GoAlexander** [comment]
The idea of this option was to set Editor (and only Editor) as system editor if the user really like it. The option with harbour-notes is for some kind of rejecting this choice.

**Comment by GoAlexander** [comment]
Not all users know about Mimer so this option is for them. 

But I understood your bug report now. Will try to fix it as soon as possible :)

**Comment by Olf0** [comment]
> But I understood your bug report now. Will try to fix it as soon as possible :)

Thanks!

**Comment by GoAlexander** [comment]
With a lag but seems that in next update it will be fixed :)
